base/sparse/abstractsparse.jl

```diff
@@ -14,3 +16,21 @@ issparse(A::AbstractArray) = false
 issparse(S::AbstractSparseArray) = true
 
 indtype{Tv,Ti}(S::AbstractSparseArray{Tv,Ti}) = Ti
+
+function union!{Tv,Ti,N}(s::AbstractSet, x::AbstractSparseArray{Tv,Ti,N})
+    vals = nonzeros(x)
+    length(vals) == length(x) ? union!(s, vals) : union!(s, [zero(Tv); vals])
+end
+
+# assumes that AbstractSparseArray type stores nonzeros in column major order
```

Comment:
I have added a comment to reflect this (as far as I can tell this is much beyond that that I can do to proactively prepare for the possibility).